### PR TITLE
Fix and enforce safe-string compatibility

### PR DESCRIPTION
**Pull request by jpdeplaix**

### CODE REVIEW

**Comment by jpdeplaix** [comment]
ping @samoht

File: gist/jbuild [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 (executables
  ((libraries (cohttp-lwt-unix github_unix cmdliner lambda-term))
-  (flags (:standard -principal -strict-sequence -g
+  (flags (:standard -principal -strict-sequence -g -safe-string
                     -warn-error "+1..47-D+49" -w "A-E-41-42-44-48"))
```
Comment (samoht) [inline gist/jbuild]: can you remove the `-warn-error` flag as well while you are fixing this kind of stuff. This is not a good idea to let this in released software.

**Comment by samoht** [comment]
I am not sure why all the runs are testing all the revdeps, this seems a bit excessive to me... Would you mind removing that variable from the `global` section, changing a 4.04.2 test into a 4.06.0 and add the `REVDEPS=true` there? Thanks!

**Comment by jpdeplaix** [comment]
Done. I used ```debian-stable``` and ```debian-testing``` for the revdeps, does that sound good ?

**Comment by samoht** [comment]
Not sure about the CI errors:

```
Command:
docker run --env-file=env.list -v /home/travis/build/mirage/ocaml-github:/repo local-build ci-opam
Unable to find image 'local-build:latest' locally
docker: Error response from daemon: pull access denied for local-build, repository does not exist or may require 'docker login'.
See 'docker run --help'.
```

but this looks like infra errors, so merging.

**Pull request merged**